src/lib.rs

```diff
@@ -94,59 +113,129 @@ fn write_mod_cmd<W: Write>(config: &ModConfig, ctx: &ConfigContext, mut writer:
+    static TASKS_DONE: Lazy<Mutex<HashSet<String>>> = Lazy::new(|| Mutex::new(HashSet::new()));
+    let cybercmd_path = Path::new(ctx.game_dir).join("tools").join("cybercmd");
+    fs::create_dir_all(&cybercmd_path).unwrap();
+    let cybercmd_path = cybercmd_path.canonicalize().unwrap();
+    let cybercmd_path = cybercmd_path.to_str().unwrap();
+    const NO_WINDOW_FLAGS: u32 = 0x08000000;
+
     for task in &config.tasks {
-        match task {
-            Task::InvokeScc {
-                path,
-                custom_cache_dir,
-                terminate_on_errors,
-            } => {
-                let cmd = Path::new(ctx.game_dir)
-                    .join("engine")
-                    .join("tools")
-                    .join("scc.exe");
-                let path = render(path, ctx.clone());
-                let custom_bundle = render(custom_cache_dir, ctx.clone());
-                const NO_WINDOW_FLAGS: u32 = 0x08000000;
-
-                let res = Command::new(&cmd)
-                    .arg("-compile")
-                    .arg(path)
-                    .arg("-customCacheDir")
-                    .arg(custom_bundle)
-                    .creation_flags(NO_WINDOW_FLAGS)
-                    .status()
-                    .ok();
-                if *terminate_on_errors == Some(true) && !matches!(res, Some(st) if st.success()) {
-                    std::process::exit(0);
+        let mut tasks_done = TASKS_DONE.lock().unwrap();
+        if tasks_done.insert(task.command.clone()) {
+            log.debug(format!("Running task: \"{}\"", task.command));
+
+            match task.command.as_str() {
+                "InvokeScc" => {
+                    let cmd = Path::new(ctx.game_dir)
+                        .join("engine")
+                        .join("tools")
+                        .join("scc.exe");
+                    let path = render(task.path.as_str(), ctx.clone());
+                    let custom_bundle = render(task.custom_cache_dir.as_str(), ctx.clone());
+
+                    let res = Command::new(&cmd)
+                        .arg("-compile")
+                        .arg(path)
+                        .arg("-customCacheDir")
+                        .arg(custom_bundle)
+                        .creation_flags(NO_WINDOW_FLAGS)
+                        .status()
+                        .ok();
+                    if task.terminate_on_errors && !matches!(res, Some(st) if st.success()) {
+                        std::process::exit(0);
+                    }
+                }
+                _ => {
+                    let cmd_path = Path::new(cybercmd_path)
+                        .join(format!("{}.exe", task.command));
+                    let cmd = cmd_path.canonicalize();
+
+                    if let Ok(cmd) = cmd {
+                        if cmd.starts_with(cybercmd_path) {
+                            let res = Command::new(&cmd)
+                                .arg("-cybercmd")
```

Comment:
I feel like this flag maybe should be in the configuration, I don't think every executable necessarily needs to know what cybercmd is

Reply:
As part of limiting execution of external commands to a specific folder, I figured the executables would probably be made specifically for it. It was an impulse based on the idea of trying to make it more secure. Your right though, I'm not sure it adds anything and handling configurable arguments may be better.